ddmrp/models/mrp_production.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+# © 2016 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# © 2016 Aleph Objects, Inc. (https://www.alephobjects.com/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, fields, models
+from .stock_warehouse_orderpoint import _PRIORITY_LEVEL
+
+
+class MrpProduction(models.Model):
+    _inherit = 'mrp.production'
+
+    @api.model
+    def _search_procurements(self):
+        return [('production_id', '=', self.id)]
+
+    @api.model
+    def _search_orderpoints(self):
+        return [('name', '=', self.move_prod_id.origin)]
+
+    @api.model
+    def _find_orderpoint_from_procurement(self, procurement):
+        orderpoint = procurement.move_dest_id.procurement_id.orderpoint_id
+        procurement = procurement.move_dest_id.procurement_id
+        return procurement, orderpoint
+
+    @api.multi
+    @api.depends('move_prod_id')
+    def _compute_orderpoint_id(self):
+        for rec in self:
+            domain = rec._search_procurements()
+            procurements = rec.env['procurement.order'].search(domain)
+            orderpoints = [procurement.orderpoint_id for procurement in
+                           procurements if procurement.orderpoint_id]
+            if orderpoints:
+                rec.orderpoint_id = orderpoints[0]
+            else:
+                for procurement in procurements:
+                    orderpoint = False
+                    originating_procurement = procurement
+                    while not orderpoint:
+                        originating_procurement, orderpoint = \
+                            self._find_orderpoint_from_procurement(
+                                originating_procurement)
+                        if orderpoint:
+                            rec.orderpoint_id = orderpoint
+                        if not originating_procurement:
+                            break
+
+    @api.multi
+    @api.depends("orderpoint_id")
+    def _compute_execution_priority(self):
+        for rec in self:
+            if rec.state not in ['done', 'cancel']:
+                rec.execution_priority_level = \
+                    rec.orderpoint_id.execution_priority_level
```

Comment:
As for the ORM definition, _rec.orderpoint_id_ is not required, so it could not exist here. Is it possible it's changed to a blank value? In that case this method would raise an exception here.